src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let amount = 0;
+
+  return function adder(x) {
+    if (!x && x !== 0) {
+      const count = amount;
+
+      amount = 0;
+
+      return count;
+    } else {
+      amount += x;
+
+      return adder;
+    }
```

Comment:
you have `return` for your if condition, so in this case you don't need to use `else`
```suggestion
    } 
    
      amount += x;

      return adder;
   
```